src/Symfony/Component/Form/Form.php

```diff
@@ -748,7 +748,7 @@ public function isEmpty()
 
         return FormUtil::isEmpty($this->modelData) ||
             // arrays, countables
-            0 === count($this->modelData) ||
+            ((is_array($this->modelData) || $this->modelData instanceof \Countable) && 0 === count($this->modelData)) ||
```

Comment:
Shouldnt this be covered `FormUtils::isEmpty` (or maybe add `isEmptyCollection`) (ie. what about https://github.com/symfony/symfony/blob/master/src/Symfony/Component/Form/Extension/Core/Type/ChoiceType.php#L98?)